packages/lesswrong/server/repos/DialogueChecksRepo.ts

```diff
@@ -58,6 +61,28 @@ class DialogueChecksRepo extends AbstractRepo<"DialogueChecks"> {
       );
     `, [userId1, userId2])
   }
+
+  async getMatchFormYourTurn(): Promise<DialogueCheckWithExtraData[]> {
+    const result = await this.getRawDb().any(`
+      SELECT 
+        dc.*,
+        dmp_reciprocal._id AS "targetUserMatchPreferenceId"
+      FROM "DialogueChecks" AS dc
+      LEFT JOIN "DialogueMatchPreferences" AS dmp ON dc._id = dmp."dialogueCheckId"
+      LEFT JOIN "DialogueChecks" AS dc_reciprocal ON dc."userId" = dc_reciprocal."targetUserId" AND dc."targetUserId" = dc_reciprocal."userId"
+      LEFT JOIN "DialogueMatchPreferences" AS dmp_reciprocal ON dc_reciprocal._id = dmp_reciprocal."dialogueCheckId"
+      LEFT JOIN "Notifications" AS n ON n."documentId" = dmp_reciprocal._id
+      WHERE 
+          dc.checked IS TRUE
+          AND dc_reciprocal.checked IS TRUE
+          AND (dmp._id IS NULL OR dmp.deleted IS TRUE)
+          AND dmp_reciprocal._id IS NOT NULL
+          AND dmp_reciprocal.deleted IS NOT TRUE 
+          AND (n._id IS NULL OR n.type != 'yourTurnMatchForm')
```

Comment:
Actually, now that I think about it, this last condition seems like it might cause us to incorrectly return dialogue checks with a `targetUserMatchPreferenceId` in the case where we've actually sent a notification for that dialogue match preference of any type other than `yourTurnMatchForm`, even if we _have_ also sent a `yourTurnMatchForm` notification for that dialogue match preference?  And the reason it's not a problem right now is that this is the only notification we have for that docment type (so far).

I don't think it's worth the effort to fix it, but to avoid this being a future footgun I'd feel a lot better if this query was just run inline in the manual migration (see `2023-07-28-viewUpdaterErrorHandling.ts` for an example of how to do this) rather than being defined as a method on the repo, with a comment above this line explaining this in case anyone is tempted to copy the query for other purposes.

Reply:
That is intended behavior. 

The intended behavior is not to send only one notification of any kind per matchPreference. It's to only send one `yourTurnMatchForm` per matchPreference -- because there might be other things we want to inform users of!